### PR TITLE
Add numpydoc configuration

### PR DESCRIPTION
**Pull request by rob-luke**
As suggested in https://github.com/mne-tools/mne-nirs/pull/280#issuecomment-856977868

Ended up finding lots of errors. Also needed fo fix warnings for pydata theme depreciations coming in 0.7 to get the green tick

### CODE REVIEW

File: mne_nirs/preprocessing/_peak_power.py [hunk 1]
```diff
@@ -35,7 +35,6 @@ def peak_power(raw, time_window=10, threshold=0.1, l_freq=0.7, h_freq=1.5,
     %(l_trans_bandwidth)s
     %(h_trans_bandwidth)s
     %(verbose)s
-
```
Comment (rob-luke) [inline mne_nirs/preprocessing/_peak_power.py:38]: @larsoner should this have been caught by the adding numpydoc_validate as you suggested?

**Comment by larsoner** [comment]
It looks like `sphinx-multiversion` is not respecting the `-nWT --keep-going` flags, because warnings are definitely there but they are not causing the build to exit with a non-zero status:

https://app.circleci.com/pipelines/github/mne-tools/mne-nirs/1169/workflows/411b8e42-fd64-4968-9b1f-72162f1aab22/jobs/1177?invite=true#step-111-1505

**Comment by rob-luke** [comment]
Thanks @larsoner, I made a new action to just run numpydoc.

I should be careful what I wish for. Current errors to be fixed are...

```text
WARNING: [numpydoc] Validation warnings while processing docstring for 'mne_nirs.experimental_design.make_first_level_design_matrix':
114
  GL02: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
115

116
  GL03: Double line break found; please use only one blank line to separate sections or paragraphs, and do not leave blank lines at the end of docstrings
117
  PR08: Parameter "add_regs" description should start with a capital letter
118
  PR06: Parameter "add_reg_names" type should use "str" instead of "string"
119
  RT04: Return value description should start with a capital letter
120

121
WARNING: [numpydoc] Validation warnings while processing docstring for 'mne_nirs.preprocessing.peak_power':
122
looking for now-outdated files... none found
123
  PR02: Unknown parameters {'times', '-------', 'Returns', 'scores'}
124
  PR07: Parameter "Returns" has no description
125
  PR07: Parameter "-------" has no description
126
  PR09: Parameter "scores" description should finish with "."
127
  RT01: No Returns section found
128

129
WARNING: [numpydoc] Validation warnings while processing docstring for 'mne_nirs.preprocessing.scalp_coupling_index_windowed':
130
  RT05: Return value description should finish with "."
131

132
WARNING: [numpydoc] Validation warnings while processing docstring for 'mne_nirs.statistics.run_GLM':
133
  PR09: Parameter "verbose" description should finish with "."
134

135
WARNING: [numpydoc] Validation warnings while processing docstring for 'mne_nirs.statistics.compute_contrast':
136
  PR01: Parameters {'glm_est'} not documented
137
  PR02: Unknown parameters {'glm_estimates'}
138

139
WARNING: [numpydoc] Validation warnings while processing docstring for 'mne_nirs.statistics.glm_region_of_interest':
140
  PR06: Parameter "cond_idx" type should use "int" instead of "integer"
141

142
WARNING: [numpydoc] Validation warnings while processing docstring for 'mne_nirs.statistics.statsmodels_to_results':
143
  PR06: Parameter "order" type should use "str" instead of "string"
144
  RT03: Return value has no description
145

146
WARNING: [numpydoc] Validation warnings while processing docstring for 'mne_nirs.visualisation.plot_nirs_source_detector':
147
  PR01: Parameters {'interaction', 'subjects_dir', 'data', 'show_axes', 'radius', 'fig', 'trans', 'verbose'} not documented
148
  PR02: Unknown parameters {'%(trans)s', '%(subjects_dir)s', 'data: array'}
149
  PR10: Parameter "data" requires a space before the colon separating the parameter name and type
150
  PR07: Parameter "%(trans)s" has no description


```

**Comment by larsoner** [comment]
Agreed it's nice just to have one quick build to check!

> I should be careful what I wish for.

Or to put it another way (my personal favorite): don't ask questions you don't want the answers to :)

But really it's hopefully 5 minutes worth of fixes of you run the test and iterate locally. I'm pretty sure it was much worse than that when I implemented an early version of those checks a while back for MNE-Python

**Comment by rob-luke** [comment]
> I'm pretty sure it was much worse than that when I implemented an early version of those checks a while back for MNE-Python

I don't want to even imagine.

@larsoner how do you deal with the errors below in MNE? 

```text
WARNING: [numpydoc] Validation warnings while processing docstring for 'mne_nirs.visualisation.plot_nirs_source_detector':
115  PR02: Unknown parameters {'%(trans)s', '%(subjects_dir)s'}
116  PR07: Parameter "%(trans)s" has no description
117   PR07: Parameter "%(subjects_dir)s" has no description
```

**Comment by rob-luke** [comment]
Ah I think it needs the @verbose in front. Will try that tomorrow.

**Comment by larsoner** [comment]
Yes, if it has a `verbose` kwarg then decorate with `@verbose`, if not then use `@fill_doc`

**Pull request merged**